code/modules/vehicles/pussywagon.dm

```diff
@@ -54,45 +50,95 @@
 	if(href_list["empty_hoover"])
 		var/obj/vehicle/train/cargo/trolley/pussywagon/PT = tow
 		PT.empty_hoover(usr)
-	if(href_list["toggle_mops"])
+	if(href_list["toggle_mop"])
 		toggle_mop(usr)
 	SSvueui.check_uis_for_change(src)
 
-/obj/vehicle/train/cargo/engine/pussywagon/attackby(obj/item/W as obj, mob/user as mob)
+/obj/vehicle/train/cargo/engine/pussywagon/attackby(obj/item/W, mob/user)
 	if(istype(W, /obj/item/key/janicart))
 		if(!key)
 			user.drop_from_inventory(W, src)
 			key = W
-			to_chat(user, SPAN_NOTICE("You slide the key into the ignition."))
+			to_chat(user, SPAN_NOTICE("You slide \the [W] into the [src]'s ignition."))
+			playsound(src, 'sound/machines/vehicles/pussywagon/key_in.ogg', 50, FALSE)
 		else
-			to_chat(user, SPAN_WARNING("\The [src] already has a key inserted."))
+			to_chat(user, SPAN_NOTICE("There is already a key in [src]'s ignition."))
 		return
 	..()
 
-/obj/vehicle/train/cargo/engine/pussywagon/turn_on()
-	if(!key)
-		audible_message("\The [src] whirrs, but the lack of a key causes the engine to shut down.")
+/obj/vehicle/train/cargo/engine/pussywagon/CtrlClick(mob/user)
+	var/list/options = list(
+		"Toggle Ignition" = image('icons/mob/screen/radial.dmi', "custodial_key"),
+		"Toggle Mopping" = image('icons/mob/screen/radial.dmi', "custodial_mop"),
+		"Toggle Vacuuming" = image('icons/mob/screen/radial.dmi', "custodial_vacuum")
+	)
+	var/chosen_option = show_radial_menu(user, src, options, radius = 42, require_near = TRUE, tooltips = TRUE)
+	if (!chosen_option)
 		return
+	switch(chosen_option)
+		if("Toggle Ignition")
+			toggle_ignition(user)
+		if("Toggle Mopping")
+			toggle_mop(user)
+		if("Toggle Vacuuming")
+			toggle_hoover(user)
+
+/obj/vehicle/train/cargo/engine/pussywagon/AltClick(mob/user)
+	if(Adjacent(user))
+		remove_key(user)
 	else
-		..()
-		update_stats()
+		return ..()
 
-/obj/vehicle/train/cargo/engine/pussywagon/turn_off()
-	..()
+/obj/vehicle/train/cargo/engine/pussywagon/remove_key(mob/user)
+	if(!key)
+		to_chat(user, SPAN_NOTICE("There is no key in \the [src]'s ignition."))
+		return
+	if(load && load != usr)
+		return
 
-	if(tow)
-		if(istype(tow, /obj/vehicle/train/cargo/trolley/pussywagon))
-			var/obj/vehicle/train/cargo/trolley/pussywagon/PW = tow
-			PW.engine_off()
+	if(ignition)
+		toggle_ignition()
+
+	user.put_in_hands(key)
+	key = null
+	to_chat(user, SPAN_NOTICE("You take out \the [W] out of [src]'s ignition."))
```

Comment:
need to pass the key item to this proc if you want to use the name

Reply:
fixed.